src/test/app/Offer_test.cpp

```diff
@@ -4090,17 +4144,28 @@ class Offer_test : public beast::unit_test::suite
         env.require (balance (bob, gwUSD(50)));
 
         // gw now requires authorization and bob has gwUSD(50).  Let's see if
-        // bob can cross alice's offer.  bob's offer shouldn't cross and
-        // alice's unauthorized offer should be deleted.
+        // bob can cross alice's offer.
+        //
+        // o With Taker bob's offer should cross alice's.
+        // o With FlowCross bob's offer shouldn't cross and alice's
+        //   unauthorized offer should be deleted.
         env (offer (bob, XRP(4000), gwUSD(40)));
         env.close();
+        std::uint32_t const bobOfferSeq = env.seq (bob) - 1;
+
+        bool const flowCross =
+            std::find (fs.begin(), fs.end(), featureFlowCross) != fs.end();
+
+        env.require (offers (bob, flowCross ? 1 : 0));
+        env.require (balance (bob, gwUSD( flowCross ? 50 : 10 )));
+
+        if (!flowCross)
+            // The rest of the test verifies FlowCross behavior.
+            return;
 
         env.require (offers (alice, 0));
         env.require (balance (alice, gwUSD(none)));
```

Comment:
I'd like to move these two lines to before the `if (!flowCross)` conditional (and of course change `alice` balance to `gwUSD(40)` if we're not in flowCross. Reasoning: it helps me orient exactly what happens in the test. In both cases alice's offer is removed. However, in one case it's removed without using it, and in the other it's used. Moving the checks before the `if` makes that clear.